pkg/volume/csi/csi_mounter.go

```diff
@@ -339,3 +339,33 @@ func removeMountDir(plug *csiPlugin, mountPath string) error {
 	}
 	return nil
 }
+
+// applyFSGroup applies the volume ownership it derives its logic
+// from https://github.com/kubernetes/kubernetes/issues/66323
+// 1) if fstype is "", then skip fsgroup (could be indication of non-block filesystem)
+// 2) if fstype is provided and pv.AccessMode == ReadWriteOnly, then apply fsgroup
+func applyFSGroup(c *csiMountMgr, fsType string, fsGroup *int64) error {
+	if fsGroup != nil {
+		if fsType == "" {
+			glog.V(4).Info(log("mounter.SetupAt WARNING: skipping fsGroup, fsType not provided"))
+			return nil
+		}
+
+		accessModes := c.spec.PersistentVolume.Spec.AccessModes
+		if !hasReadWriteOnce(accessModes) {
```

Comment:
AccessModes in PV spec is expected to be the `all ways` the volume can support. Should we get this info from PVC spec ?

Reply:
@NickrenREN I will take a look at that to see if the code has access to the PVC at that point.

Reply:
yes, dynamic provisioning PVs' AccessModes are set to PVC requested AccessModes now, but i  don't think this is appropriate according to the PV/PVC Spec AccessModes comments.
And also, this is not suitable for pre-created PVs.
To clarify, this is not CSI specific, all volume plugins have this problem.

Reply:
>yes, dynamic provisioning PVs' AccessModes are set to PVC requested AccessModes now, but i don't think this is appropriate according to the PV/PVC Spec AccessModes comments.

@NickrenREN to clarify, are you OK with getting AccessModes from PV?  As you pointed out, getting it from PVC would not work for pre-created volumes.

Reply:
yes, today, all plugins are getting AccessModes from PV, i'm ok with this now.
I will try to figure out if we could fix this later in a separate PR.